examples/update-states.js

```diff
@@ -0,0 +1,103 @@
+const async = require('async');
+const Redis = require('ioredis');
+
+/*
+This script updates the state of Utapi `numberOfObjects` and `storageUtilized`
+starting from the current timestamp (i.e., the latest 15 minute interval).
+
+To use:
+* Set REDIS_ENDPOINT to the host and port where the Redis server is running
+* Set the various metric level states (BUCKET_STATES, ACCOUNT_STATES,
+  USER_STATES, SERVICE_STATE) to your desired values. This can be an array of n
+  length of objects. Each value `storageUtilized` and `numberOfObjects` of the
+  state objects is optional, and you should only include those that you want to
+  change. To skip updating a particular metric level state, the state should be
+  set to an empty array.
+*/
+
+const REDIS_ENDPOINT = {
+    host: '127.0.0.1',
+    port: 6379,
+};
+const BUCKET_STATES = [
+    {
+        bucket: '<bucket-name>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    },
+];
+const ACCOUNT_STATES = [
+    {
+        accountId: '<canonical-id>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    },
+];
+const USER_STATES = [
+    {
+        userId: '<user-id>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    },
+];
+const SERVICE_STATE = [
+    {
+        service: '<service-name>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    },
+];
+
+const allStates = {
+    buckets: BUCKET_STATES,
+    accounts: ACCOUNT_STATES,
+    users: USER_STATES,
+    service: SERVICE_STATE,
+};
+
+const allParams = [];
+Object.keys(allStates).forEach(key => allStates[key].forEach(state =>
```

Comment:
For an example processing, I would think it's better to just declare the final `allParams` array statically instead of post-processing the various exploded arrays, even though there is some redundancy, because the reader will have less steps to understand what is actually provided to redis API. The user is likely to construct the redis commands his own way to match his inputs, which may not be similar to the structures here.

Reply:
Updated to use a single array.